pkg/bootkube/bootkube.go

```diff
@@ -106,5 +111,46 @@ func (b *bootkube) Run() error {
 	go func() { errch <- WaitUntilPodsRunning(requiredPods, assetTimeout) }()
 
 	// If any of the bootkube services exit, it means it is unrecoverable and we should exit.
-	return <-errch
+	err := <-errch
+	if err != nil {
+		return err
+	}
+
+	// Now that bootkube has finished without an error,
+	// move our "user space" checkpointing pod into place.
+	return ioutil.WriteFile(checkpointPodPath, checkpointPod, 0644)
+}
+
+var checkpointPod = []byte(`
+{
```

Comment:
I feel like there is a more portable way of distributing this.

What about the checkpoint pod knows how to install itself as a static pod. For example:

We deploy a checkpoint-install pod the same way we deploy any other object (daemonset template generated by render, injected via api). The checkpoint-install pod is really just using the same checkpoint image.. just referencing internal manifest files rather than the binary itself:

The checkpoint-installer container spec has a command: "/install.sh":

```
#!/bin/bash
cp /checkpoint.yaml /kubernetes/manifests
while true;
  sleep 3600
done
```

Then it has mounts for

host=/etc/kubernetes/manifests
dest=/kubernetes/manifests

This way it basically knows how to update itself.. just drops it's files in place and will survive reboots. If a new version comes along (for whatever reason) it will just replace the existing checkpointer static pod on disk.